server/src/test/java/org/elasticsearch/action/search/ParsedScrollIdTests.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.search;
+
+import org.elasticsearch.search.internal.ShardSearchContextId;
+import org.elasticsearch.test.ESTestCase;
+
+public class ParsedScrollIdTests extends ESTestCase {
+    public void testHasLocalIndices() {
+        final int nResults = randomIntBetween(1, 3);
+        final SearchContextIdForNode[] searchContextIdForNodes = new SearchContextIdForNode[nResults];
+
+        boolean hasLocal = false;
+        for (int i = 0; i < nResults; i++) {
+
+            String clusterAlias = randomBoolean() ? randomAlphaOfLength(8) : null;
```

Comment:
ditto https://github.com/elastic/elasticsearch/pull/62053/files#r487806064

Reply:
I tried to remove the randomness, but then decided to keep it as is. Here is why: The intention is to test that "any local indices will be reported as hasLocalIndices". The "any" bit makes it more suitable for randomness, because it can be:
* A single search context
* Multiple search contexts where 
  * the local one can be in the first position, second position, last position etc. 
  * Or there might be more than one local contexts, e.g. all of them are local contexts or only one of them is not local context.

The randomness in the test is intended to cover all above situations. If removed, the coverage will be reduced (or we can replace it with a very bloated test method, but I think that is less ideal).